packages/ui5-application-writer/src/index.ts

```diff
@@ -50,12 +50,19 @@ async function generate(basePath: string, ui5AppConfig: Ui5App, fs?: Editor): Pr
     // ui5-local.yaml
     const ui5LocalConfigPath = join(basePath, 'ui5-local.yaml');
     const ui5LocalConfig = await UI5Config.newInstance(fs.read(ui5LocalConfigPath));
-    ui5LocalConfig.addUI5Framework(
-        ui5App.ui5.framework,
-        ui5App.ui5.localVersion,
-        ui5App.ui5.ui5Libs as string[],
-        ui5App.ui5.ui5Theme
-    );
+    const ui5Libs = ui5App.ui5.ui5Libs as string[];
+    const ui5ReuseLibs = ui5App.ui5.ui5ReuseLibs;
+
+    //remove reuse libs
+    if (ui5ReuseLibs) {
+        for (let i = 0; i < ui5Libs.length; i++) {
+            if (ui5ReuseLibs.indexOf(ui5Libs[i]) > -1) {
+                ui5Libs.splice(i, 1);
+            }
+        }
+    }
+
+    ui5LocalConfig.addUI5Framework(ui5App.ui5.framework, ui5App.ui5.localVersion, ui5Libs, ui5App.ui5.ui5Theme);
```

Comment:
Why not pass the `ui5Libs` directly now since they are distinct from the `ui5ReuseLibs` as they should be treated differently?:
```
ui5LocalConfig.addUI5Framework(ui5App.ui5.framework, ui5App.ui5.localVersion, ui5Libs, ui5App.ui5.ui5Theme);
```